### PR TITLE
Include whether or not we're using GPUs in command/stats

### PR DESCRIPTION
**Pull request by billyb2**
### Change Summary

What and Why:
WE're now including whether or not a command is using a GPU in metrics, so we can more easily do some useful queries.

How:
We just check fly.toml to see if it contains gpu_kind anywhere. This should work for every command, but it'll be especially useful for deploys

Related to:

https://github.com/superfly/flyctl-metrics/pull/23

---

### Documentation

- [ ] Fresh Produce
- [ ] In superfly/docs, or asked for help from docs team
- [ ] n/a

### CODE REVIEW

File: internal/command/command.go [hunk 1]
```diff
@@ -694,3 +696,22 @@ func ChangeWorkingDirectory(ctx context.Context, wd string) (context.Context, er
...
+func setUsingGPU(ctx context.Context) (context.Context, error) {
+	appConfig := appconfig.ConfigFromContext(ctx)
+	if appConfig == nil {
+		instrument.UsingGPU = false
+		return ctx, nil
+	}
+
+	usingGPU := false
+	for _, compute := range appConfig.Compute {
+		if compute != nil && compute.GPUKind != "" {
```
[the remark this answers is not in the document]
Comment (dangra) [reply internal/command/command.go:709]: alternatively you can:

```go
instrument.UserGPU = lo.SomeBy(appConfig.Compute, func(x *appconfig.Compute) bool {
  return x != nil && x.MachineGuest != nil && x.MachineGuest.GPUKind != ""
})
```

**Review by dangra — CHANGES_REQUESTED** [review]

File: internal/cli/cli.go [hunk 2]
```diff
@@ -86,7 +86,7 @@ func Run(ctx context.Context, io *iostreams.IOStreams, args ...string) int {
 
 	cmd, err = cmd.ExecuteContextC(ctx)
 
-	if err != nil {
+	if cmd != nil {
```
[the remark this answers is not in the document]
Comment (billyb2) [reply internal/cli/cli.go:89]: And we want to always show that a command finish, even if it errored. We should probably incldue whether or not the command finishing was an error or not in command/stats though

**Review by dangra — APPROVED** [review]

File: internal/command/command.go [hunk 3]
```diff
@@ -694,3 +697,17 @@ func ChangeWorkingDirectory(ctx context.Context, wd string) (context.Context, er
...
+func setUsingGPU(ctx context.Context) (context.Context, error) {
+	appConfig := appconfig.ConfigFromContext(ctx)
+	if appConfig == nil {
+		instrument.UsingGPU = false
+		return ctx, nil
+	}
+
+	instrument.UsingGPU = lo.SomeBy(appConfig.Compute, func(x *appconfig.Compute) bool {
+		return x != nil && x.MachineGuest != nil && x.MachineGuest.GPUKind != ""
+	})
```
Comment (dangra) [inline internal/command/command.go:710]: ignore me but had to suggest a simplification nitpick:

```suggestion
	if appConfig != nil {
		instrument.UsingGPU = lo.SomeBy(appConfig.Compute, func(x *appconfig.Compute) bool {
			return x != nil && x.MachineGuest != nil && x.MachineGuest.GPUKind != ""
		})
	}
```

File: internal/command/command.go [hunk 4]
```diff
@@ -694,3 +697,18 @@ func ChangeWorkingDirectory(ctx context.Context, wd string) (context.Context, er
...
+	appConfig := appconfig.ConfigFromContext(ctx)
+	if appConfig != nil {
+		instrument.UsingGPU = lo.SomeBy(appConfig.Compute, func(x *appconfig.Compute) bool {
+			return x != nil && x.MachineGuest != nil && x.MachineGuest.GPUKind != ""
+		})
+	}
+
+	instrument.UsingGPU = lo.SomeBy(appConfig.Compute, func(x *appconfig.Compute) bool {
+		return x != nil && x.MachineGuest != nil && x.MachineGuest.GPUKind != ""
+	})
```
Comment (dangra) [inline internal/command/command.go:711]: remove this

Comment (billyb2) [reply internal/command/command.go]: forgot to stage removing this :(

**Pull request merged**